api/src/opentrons/protocol_api/core/engine/instrument.py

```
@@ -1237,7 +1238,158 @@ def consolidate_liquid(
         tip_racks: List[Tuple[Location, LabwareCore]],
         trash_location: Union[Location, TrashBin, WasteChute],
     ) -> None:
-        pass
+        if not tip_racks:
+            raise RuntimeError(
+                "No tipracks found for pipette in order to perform transfer"
+            )
+        tiprack_uri_for_transfer_props = tip_racks[0][1].get_uri()
+        transfer_props = liquid_class.get_for(
+            pipette=self.get_pipette_name(), tip_rack=tiprack_uri_for_transfer_props
+        )
+        blow_out_properties = transfer_props.dispense.retract.blowout
+        if (
+            blow_out_properties.enabled
+            and blow_out_properties.location == BlowoutLocation.SOURCE
+        ):
+            raise RuntimeError(
+                'Blowout location "source" incompatible with consolidate liquid.'
+                ' Please choose "destination" or "trash".'
+            )
```

We should highlight this and mix, pre-wet behavior in docs. I guess it's okay to raise this error instead of simply skipping the blowout, as long as our built-in classes don't have blowout with source location enabled by default, because otherwise a lot of protocols will run into errors due to no fault of the users. Liquid classes are just supposed to work without having to modify for each transfer.

---

Did an inventory of the three liquid classes and all have blowout with enabled set to False, with no parameters set. So in order to get this to raise, the user would have to specifically set the location to be `SOURCE` and enable blowout.
The reason why I've set this to be an error is that unlike pre-wet or pre-aspirate mix, we're not unconditionally skipping the step, rather we are unable to do one specific combination of settings, and there is no good decision on whether to skip it entirely or change the blowout location, so I'd rather just raise and force the user to decide what change to make if they get themselves in this state.